### PR TITLE
feat/reset-password

### PR DESCRIPTION
**Pull request by IgorVieira**
image
image
image
image
image
image

### CODE REVIEW

**Comment by paulchrisluke** [comment]
image

@IgorVieira On Desktop, please make all content fit within our content margin, See our "new 
Memo" flow on staging.thebluedelta.com for reference 

image

File: src/components/AuthLogin.vue [hunk 1]
```diff
@@ -149,3 +158,14 @@ export default {
   }
 };
 </script>
+<style scoped>
+.auth-button {
```
Comment (mort3za) [inline src/components/AuthLogin.vue:162]: We prefer tailwind classes over custom styles. Please use them as much as possible.

File: src/store/modules/auth.js [hunk 2]
```diff
@@ -196,9 +224,55 @@ export default {
...
+        returnUrl: `${host}/new-password`
+      };
+
+      const configs = {
+        method: 'post',
+        url,
+        data: body
+      };
+
+      ajax(configs)
```
Comment (mort3za) [inline src/store/modules/auth.js:242]: Please use meredith-axios package for ajax requests, you can find similar requests in register or login methods.

File: src/views/NewPassword.vue [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  width: 184px;
+  outline: none;
+  text-align: center;
+  letter-spacing: 0.02em;
+  text-transform: uppercase;
+  font-size: 14px;
+}
+
+.reset-title {
+  font-family: Work Sans;
```
Comment (mort3za) [inline src/views/NewPassword.vue:203]: We have typography styles ready to use in typography.css file, no need to redefine them.

File: src/views/NewPassword.vue [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+      });
+    }
+  },
+  computed: {
+    passwordMatch() {
+      return this.confirmPassword !== this.newPassword;
+    },
+    newPassword: {
+      get() {
+        return this.$store.getters['auth/newPassword'];
```
Comment (mort3za) [inline src/views/NewPassword.vue:121]: This is OK, however using mapGetters and mapMutations is easier to write/read.

File: src/views/VolkswagenLoadMessage.vue [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  justify-content: center;
+}
+
+.loading-message {
+  font-weight: 500;
+  font-size: 20px;
+  line-height: 25px;
+  margin-top: 28px;
+  margin-right: 33px;
+  margin-left: 33px;
```
Comment (mort3za) [inline src/views/VolkswagenLoadMessage.vue:64]: I didn't check UI, but number 33 in styles should not be seen by 99.99%
There should be something wrong. good values are a multiply of 8.

File: src/views/VolkswagenLoadMessage.vue [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    newPath() {
+      return this.redirect();
+    }
+  }
+};
+</script>
+
+<style scoped>
+.loading {
+  background: radial-gradient(50% 49.2% at 0% 0.8%, rgba(255, 255, 255, 0.25) 0%, rgba(255, 255, 255, 0) 100%), #011d51;
```
Comment (mort3za) [inline src/views/VolkswagenLoadMessage.vue:49]: If this is not the same bg-brand-gradient, please add it beside it in tailwind config.

File: src/store/modules/auth.js (hunk 2 above)
Comment (IgorVieira) [reply src/store/modules/auth.js:242]: We don't have something like that for reset password in meredith-axios, but I have opened a pull request for add: https://github.com/whynotearth/meredith-axios/pull/2

File: src/views/NewPassword.vue (hunk 4 above)
Comment (IgorVieira) [reply src/views/NewPassword.vue:121]: I'm not very familiar using mapGetters our mapMutations, sorry about that 😔

File: src/views/VolkswagenLoadMessage.vue (hunk 5 above)
Comment (IgorVieira) [reply src/views/VolkswagenLoadMessage.vue:64]: I have changed

File: src/views/VolkswagenLoadMessage.vue (hunk 6 above)
Comment (IgorVieira) [reply src/views/VolkswagenLoadMessage.vue:49]: Solved

File: src/components/AuthLogin.vue (hunk 1 above)
Comment (IgorVieira) [reply src/components/AuthLogin.vue:162]: Okay, thank you

File: src/store/modules/auth.js (hunk 2 above)
Comment (mort3za) [reply src/store/modules/auth.js]: You're right, I just updated meredith-axios with the latest API routes. :+1:

File: .env [hunk 7]
```diff
@@ -2,4 +2,5 @@
 # VUE_APP_DEV_MODE=1
 VUE_APP_TITLE="VW Emails Corporate Communications CMS"
 VUE_APP_COMPANY_SLUG="volkswagen"
+VUE_APP_HOST=https://staging.thebluedelta.com/
```
Comment (mort3za) [inline .env:5]: Missed this one, you should use window.location.origin instead. No need to define a host variable.

**Comment by mort3za** [comment]
I'm checking and fix this PR

**Pull request merged**